src/main/java/org/folio/rest/converter/providers/ProviderPutRequestConverter.java

```diff
@@ -24,7 +23,8 @@ public VendorPut convert(@NonNull ProviderPutRequest provider) {
     // value
     // Hard code it to false; it should not affect the state of inherited that RM
     // API maintains
-    if (provider.getData().getAttributes().getProxy() != null) {
+    if (provider.getData().getAttributes().getProxy() != null &&
```

Comment:
When updating provider, Ruby module actually retrieves old provider data and uses it if proxy or token are not set.
I think it is not necessary, if proxy or token are set, then UI will send them, and if it is not set, then we will just send null to RM API.